src/AppInstallerCommonCore/Manifest/ManifestYamlPopulator.cpp

```diff
@@ -1083,20 +1152,95 @@ namespace AppInstaller::Manifest
                 ManifestLocalization localization;
                 auto errors = ValidateAndProcessFields(entry, LocalizationFieldInfos, VariantManifestPtr(&localization));
                 std::move(errors.begin(), errors.end(), std::inserter(resultErrors, resultErrors.end()));
-                manifest.Localizations.emplace_back(std::move(std::move(localization)));
+                m_manifest.get().Localizations.emplace_back(std::move(std::move(localization)));
+            }
+        }
+
+        return resultErrors;
+    }
+
+    ValidationErrors ManifestYamlPopulator::InsertShadow(const YAML::Node& shadowNode)
+    {
+        Manifest shadowManifest;
+
+        // Process shadow node.
+        auto resultErrors = ValidateAndProcessFields(shadowNode, GetShadowRootFieldProcessInfo(), VariantManifestPtr(&shadowManifest));
+
+        // Merge.
+        if (m_manifestVersion.get() >= ManifestVer{ s_ManifestVersionV1_5 })
+        {
+            // Default localization
+            if (m_manifest.get().DefaultLocalization.Locale == shadowManifest.DefaultLocalization.Locale)
+            {
+                // Icons
+                if (!m_manifest.get().DefaultLocalization.Contains(Localization::Icons) &&
+                    shadowManifest.DefaultLocalization.Contains(Localization::Icons))
+                {
+                    m_manifest.get().DefaultLocalization.Add<Localization::Icons>(std::move(shadowManifest.DefaultLocalization.Get<Localization::Icons>()));
+
+                    YAML::Node key{ YAML::Node::Type::Scalar, "", YAML::Mark() };
+                    key.SetScalar("Icons");
+                    YAML::Node value = shadowNode["Icons"];

```

Comment:
We probably want case insensitivity here, but I think you might have to add that.